lib/packetgen/types/fields.rb

```diff
@@ -138,11 +138,13 @@ def self.define_field(name, type, options={})
              type.instance_methods.include? :from_human
           define << "def #{name}; self[:#{name}].to_human; end"
           define << "def #{name}=(val) self[:#{name}].from_human val; end"
+        elsif type == Types::HTTPHeaders
+          define << "def #{name}; self[:#{name}].data; end\n"
+          define << "def #{name}=(val) self[:#{name}].read val; end"
         else
```

Comment:
I prefer HTTHeaders to not be a type, so you should not modify Fields.define_fields: prefer using #to_human and #from_human (already used in DNS::Name, IP::Addr or IPv6::Addr, by example).